code/game/objects/structures/altar.dm

```diff
@@ -0,0 +1,80 @@
+/*
+ *Altars and Podiums
+ *Altars should later have other functions but I am not coder enough to do them yet.
+ *
+ */
+
+/obj/structure/altar
+	name = "altar"
+	desc = "Some sort of spooky altar used by the chaplain for, something. Best not to fiddle with it."
+	icon = 'icons/obj/DeusSpritus.dmi'
+	icon_state = "altar2_on"
+	density = 1
+	anchored = 1
+	var/on = 1
+	light_range = 4
+	light_color = LIGHT_COLOR_ORANGE
+	var/last_check = 0
+
+/obj/structure/altar/New()
+	..()
+	processing_objects.Add(src)
+
+/obj/structure/altar/Destroy()
+	processing_objects.Remove(src)
+	..()
+
+
+/obj/structure/altar/attack_hand(mob/M as mob)
```

Comment:
You can write this more elegantly as

`on = !on
set_light(on*4)
update_icon()`

With no if statement